src/ui/ui_exports.js

```diff
@@ -48,15 +49,21 @@ class UiExports {
       case 'app':
       case 'apps':
         return (plugin, specs) => {
+          const id = plugin.id;
           for (let spec of [].concat(specs || [])) {
-            let app = this.apps.new(_.defaults({}, spec, {
-              id: plugin.id,
-              urlBasePath: this.urlBasePath
-            }));
+            const app = this.apps.new({ id, ...spec });
```

Comment:
I'm not entirely sure what can make up `spec`.  Is it possible for people to have an `id` property in `spec`?  If so, perhaps we should define our `id` property *after* `...spec` so it can't accidentally be overwritten?